src/server/lock_manager/client.rs

```diff
@@ -60,12 +60,16 @@ impl Client {
         let send_task = Box::pin(async move {
             let mut sink = sink.sink_map_err(Error::Grpc);
 
-            sink.send_all(&mut rx.map(|r| Ok((r, WriteFlags::default()))))
-                .await
-                .map(|_| {
-                    info!("cancel detect sender");
-                    sink.get_mut().cancel();
-                })
+            sink.send_all(&mut rx.map(|r| {
+                // update metrics
+                DETECTOR_SEND_CHANNEL_SIZE_GAUGE.dec();
+                Ok((r, WriteFlags::default()))
```

Comment:
`DETECTOR_SEND_CHANNEL_SIZE_GAUGE` only tells us the channel size, but the max throughput of the sink is unknown. How about to use two counnters (eg. `queued` & `sent`), then the channel size is approximately equal to `queued - sent`, and the max throughput is about `rate(sent)` when `rate(sent) < rate(queued)`.

Reply:
Good idea